src/Mediator.Switch.Extensions.Microsoft.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -36,7 +36,15 @@ public static IServiceCollection AddScoped<TSwitchMediator>(this IServiceCollect
 
             foreach (var handlerType in notificationHandlerTypes)
             {
+                // Register the concrete type
                 services.AddScoped(handlerType);
+                
+                // Also register against notification handler interfaces
+                foreach (var handlerInterface in handlerType.GetInterfaces()
+                    .Where(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(INotificationHandler<>)))
+                {
+                    services.AddScoped(handlerInterface, handlerType);
```

Comment:
I wonder if we should replicate the way handlerType's registered in `OrderNotificationHandlers` where it registers the interface against the same concrete class instance, i.e. `services.AddScoped(handlerInterface, sp => sp.GetRequiredService(handlerType));` within the same scope?